src/table-view/table-view.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import createReactClass from 'create-react-class'
+import PropTypes from 'prop-types'
+
+import { default as Table } from '../table'
+import { default as Select } from '../select'
+import { default as Button } from '../button'
+
+export default createReactClass({
+  displayName: 'TableView',
+  propTypes: {
+    rows: PropTypes.array,
+    columns: PropTypes.array,
+    filters: PropTypes.array,
+    onFilterChange: PropTypes.func,
+    onClearFilters: PropTypes.func
+  },
+  getDefaultProps() {
+    return {
+      rows: [],
+      columns: [],
+      filters: []
+    }
+  },
+
+  render() {
+    const { rows, columns, onFilterChange, onClearFilters } = this.props
+    const filters = sanitizeFilters(this.props.filters)
```

Comment:
I have mentioned one alternative [in previous comment](https://github.com/jayesh-bhoot/challenge-table-view/pull/1#discussion_r309991832) along with its caveats.

Within the confines of the existing API design (stateless `TableView`), these values should be regulated in the parent when being passed as props.

The last alternative, as discussed [here](https://github.com/jayesh-bhoot/challenge-table-view/pull/1#discussion_r309992180), is memoization.

Though I have a feeling that I am missing something subtle here. If so, please elaborate.

Reply:
these solutions will work and solve the problem but they'll also add more complexity.
what do you think of just doing an in-line existence check?
```js
  value={f.value || [ ] }
```

Reply:
This is embarassing. I guess I jumped on a track and kept going in that direction!

I should have revisited the code and may be I would have caught your solution.

Reply:
Actually looking with a fresh mind at why I didn't go with your recommendation in the first place, the `filters = sanitizeFilters()` provides the sanitized filters for all subsequent operations, not just in populating `select`s.

I would rather have a single-place sanitization than having to litter the code with the same `f.value || []` check.

Anyway, I am moving the filter control in `TableView` so that should nullify this issue.